### PR TITLE
feat: resources added to project

### PR DESCRIPTION
**Pull request by benstoltz**
1. Description: Expands Project/ lays the groundworks to adding/updating/fetching of specific resources in projects/other item entities in the future.

1. Instructions for testing:

1. Closes Issues: #<number> (if appropriate)

1. [x] used semantic commit messages
  
1. [x] PR title follows semantic commit format (**CRITICAL** if the title is not in a semantic format, the release automation will not run!)

### CODE REVIEW

File: packages/common/src/projects/edit.ts [hunk 1]
```diff
@@ -153,3 +182,31 @@ export async function getHubProjectEditorConfig(
...
+ */
+/* istanbul ignore next */
+function configureProjectResources(resources: any) {
+  // Set up obj tracking resource prop && filename (KEEP IN SYNC WITH getPropertyMap()
+  // for additional resources!)
+  const resourceNameAndProp: {
+    [key: string]: string;
+  } = {
+    location: "location.json",
+  };
```
Comment (benstoltz) [inline packages/common/src/projects/edit.ts:201]: Per @dbouwman we want to be very explicit about prop/filename matching. We should consider a centralized location to keep this obj, the one in fetch in sync along with the prop names in getBasePropertyMap.

File: packages/common/src/projects/fetch.ts [hunk 2]
```diff
@@ -55,6 +55,19 @@ export async function convertItemToProject(
   requestOptions: IRequestOptions
 ): Promise<IHubProject> {
   const model = await fetchModelFromItem(item, requestOptions);
+  // Set up obj tracking resource prop && filename (KEEP IN SYNC WITH getPropertyMap()
+  // for additional resources!)
+  const resourceNameAndPropPairs: {
+    [key: string]: string;
+  } = {
+    location: "location.json",
+  };
```
Comment (benstoltz) [inline packages/common/src/projects/fetch.ts:64]: Per @dbouwman we want to be very explicit about prop/filename matching. We should consider a centralized location to keep this obj, the one in edit in sync along with the prop names in getBasePropertyMap.

File: packages/common/src/core/types/IHubLocation.ts [hunk 3]
```diff
@@ -0,0 +1,26 @@
+import { ISpatialReference } from "@esri/arcgis-rest-types";
+
+export interface IHubLocation {
```
Comment (dbouwman) [inline packages/common/src/core/types/IHubLocation.ts:3]: jsdoc comment?

Comment (dbouwman) [reply packages/common/src/core/types/IHubLocation.ts:3]: Also - please run `npm run docs:serve` and verify that these property comments are included - I'm not 100% sure if they pick up single-line comments `//` vs the `/** ...` style

File: packages/common/src/models/index.ts [hunk 4]
```diff
@@ -163,6 +166,84 @@ export function updateModel(
...
+ *     resource: any;
+ *     filename: string;
+ *   }>} resources
+ * @param {IUserRequestOptions} requestOptions
+ * @return {*}  {Promise<IModel>}
+ */
+export async function upsertModelResources(
+  model: IModel,
+  resources: Array<{
+    resource: any;
```
Comment (dbouwman) [inline packages/common/src/models/index.ts:186]: I think this should be typed `Record<string any>` vs `any` b/c the latter allows `null` and `undefined`

File: packages/common/src/models/index.ts [hunk 5]
```diff
@@ -180,6 +261,39 @@ export async function fetchModelFromItem(
...
+ * @param {IRequestOptions} requestOptions
+ * @return {*}  {Promise<any>}
+ */
+export async function fetchModelResources(
+  item: IItem,
+  resourceNamePairs: {
+    [key: string]: string;
+  },
+  requestOptions: IRequestOptions
+): Promise<any> {
```
Comment (dbouwman) [inline packages/common/src/models/index.ts:282]: same with this `any` -> `Record<string, any>`

**Review by dbouwman — APPROVED** [review]
Minor changes re: docs and some types

**Pull request merged**